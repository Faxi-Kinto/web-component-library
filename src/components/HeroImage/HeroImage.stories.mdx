import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
import HeroImage from './';

<Meta title="Core|Atoms/HeroImage" />

# HeroImage Component

Image component designed to fill its container.

## Implementation

<Preview>
  <Story name="withImage">
    <div style={{ height: '100vh' }}>
      <HeroImage url="assets/images/create-community-form/KINTO_images_step-1.jpg" />
    </div>
  </Story>
</Preview>

## Props

<Props of={HeroImage} />
