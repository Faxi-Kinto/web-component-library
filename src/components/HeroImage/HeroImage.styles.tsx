import styled from 'styled-components';

type BackgroundProps = {
  url: string;
};

export const Background = styled.div`
  height: 100%;
  background: ${(props: BackgroundProps) => `
    url('${props.url}') no-repeat center center fixed`};
  background-size: cover;
`;
