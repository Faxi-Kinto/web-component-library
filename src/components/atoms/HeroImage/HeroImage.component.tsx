import React from 'react';
import * as Styled from './HeroImage.styles';

/**
 * @name HeroImage
 *
 * Displays an image by returning a div with an image-filled background
 *
 * @returns {JSX}
 */

export type HeroImageProps = {
  url: string;
  className?: string;
};

const HeroImage: React.FC<HeroImageProps> = (
  props: HeroImageProps
): JSX.Element => {
  const { url, className } = props;

  return <Styled.Background url={url} className={className} />;
};

export default HeroImage;
