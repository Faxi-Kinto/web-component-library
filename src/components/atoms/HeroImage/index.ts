import HeroImage from './HeroImage.component';

export default HeroImage;
