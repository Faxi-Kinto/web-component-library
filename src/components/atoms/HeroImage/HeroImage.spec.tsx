import React from 'react';
import HeroImage from '.';
import renderer from 'react-test-renderer';
import { HeroImageProps } from './HeroImage.component';

test('HeroImage renders correctly', () => {
  const props: HeroImageProps = {
    url: '',
  };
  const tree = renderer.create(<HeroImage {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
