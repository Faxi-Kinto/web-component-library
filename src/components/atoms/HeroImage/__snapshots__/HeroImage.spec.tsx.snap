// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroImage renders correctly 1`] = `
.c0 {
  height: 100%;
  background: url('') no-repeat center center fixed;
  background-size: cover;
}

<div
  className="c0"
/>
`;
